src/load-script.ts

```diff
@@ -80,8 +85,32 @@ export function loadCustomScript(
             url,
             attributes,
             onSuccess: () => resolve(),
-            onError: () =>
-                reject(new Error(`The script "${url}" failed to load.`)),
+            onError: () => {
+                const defaultError = new Error(
+                    `The script "${url}" failed to load.`
+                );
+
+                if (!window.fetch) {
+                    return reject(defaultError);
+                }
+                // attempt to fetch() the error reason from the response body
+                return fetch(url)
+                    .then((response) => {
```

Comment:
Let's only do this when `response.status !== 200`

Reply:
Quick question here onError callback will happen only when the request is not a 200 HTPP status?